api/src/Model/Services/UserData.php

```diff
@@ -103,7 +103,7 @@ function getUsers($getCount, $isStaffMember, $stringMatch, $page, $sortBy = null
         $extc = '';
         $group = 'GROUP BY p.personid';
 
-        if (($personId && !$isManager) || $pid || (!$isStaffMember && !$visitName))
+        if ((($personId && !$isManager) || (!$isStaffMember && !$visitName)) && $pid)
```

Comment:
I think this is not right look at https://github.com/DiamondLightSource/SynchWeb/blob/82b6faa413bafd1d1b535c7765a094717964a177/api/src/Page/Users.php#LL185C29-L185C32